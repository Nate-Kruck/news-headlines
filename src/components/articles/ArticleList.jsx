import React from 'react';
import PropTypes from 'prop-types';
import styles from './Articles.css';
import Article from '../Article/Article';

function ArticleList({ articles }) {
  const ArticleElements = articles.map(article => (
    <Article key={article.title} article={article}/>
  ));

  return (
    <div className={styles.articles}>
      <ul>{ArticleElements}</ul>
    </div>
  );
}

ArticleList.propTypes = {
  articles: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string,
    author: PropTypes.string,
    description: PropTypes.string,
    imageUrl: PropTypes.string,
    url: PropTypes.string
  })).isRequired
};

export default ArticleList;
