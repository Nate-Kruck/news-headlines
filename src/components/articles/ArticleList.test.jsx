import React from 'react';
import { 
  render,
  cleanup,
  screen
} from '@testing-library/react';
import ArticleList from './ArticleList';

describe('ArticleList Testing', () => {
  afterEach(() => cleanup());
  
  it('render an ArticleList', () => {
    const articles = [
      {
        title: 'title1',
        author: 'Bertholomule',
        description: 'Hey friends',
        imageUrl: 'http://placeholder.com/articles',
        url: 'nate-kruck.com',
      },
      {
        title: 'title2',
        author: 'Bertholomule',
        description: 'Hey friends',
        imageUrl: 'http://placeholder.com/articles',
        url: 'nate-kruck.com',
      },
      {
        title: 'title3',
        author: 'Bertholomule',
        description: 'Hey friends',
        imageUrl: 'http://placeholder.com/articles',
        url: 'nate-kruck.com',
      },
    ];
    const { asFragment } = render(<ArticleList articles={articles} />);

    const allDescriptions = screen.getAllByText('Hey friends');

    expect(allDescriptions.length).toEqual(3);
  
    expect(asFragment()).toMatchSnapshot();
  });
});
