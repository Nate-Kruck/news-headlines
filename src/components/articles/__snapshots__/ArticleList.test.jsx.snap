// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList Testing render an ArticleList 1`] = `
<DocumentFragment>
  <div
    class="articles"
  >
    <ul>
      <div
        class="mainGrid"
      >
        <li
          class="Article"
        >
          <span
            class="title"
          >
            <h1
              data-testid="article"
            >
              title1
            </h1>
          </span>
          <span
            class="imageContainer"
          >
            <a
              href="nate-kruck.com"
              rel="noreferrer"
              target="_blank"
            >
              <img
                alt="title1"
                src="http://placeholder.com/articles"
              />
            </a>
          </span>
          <p
            class="author"
          >
            Bertholomule
          </p>
          <p
            class="description"
          >
            Hey friends
          </p>
        </li>
      </div>
      <div
        class="mainGrid"
      >
        <li
          class="Article"
        >
          <span
            class="title"
          >
            <h1
              data-testid="article"
            >
              title2
            </h1>
          </span>
          <span
            class="imageContainer"
          >
            <a
              href="nate-kruck.com"
              rel="noreferrer"
              target="_blank"
            >
              <img
                alt="title2"
                src="http://placeholder.com/articles"
              />
            </a>
          </span>
          <p
            class="author"
          >
            Bertholomule
          </p>
          <p
            class="description"
          >
            Hey friends
          </p>
        </li>
      </div>
      <div
        class="mainGrid"
      >
        <li
          class="Article"
        >
          <span
            class="title"
          >
            <h1
              data-testid="article"
            >
              title3
            </h1>
          </span>
          <span
            class="imageContainer"
          >
            <a
              href="nate-kruck.com"
              rel="noreferrer"
              target="_blank"
            >
              <img
                alt="title3"
                src="http://placeholder.com/articles"
              />
            </a>
          </span>
          <p
            class="author"
          >
            Bertholomule
          </p>
          <p
            class="description"
          >
            Hey friends
          </p>
        </li>
      </div>
    </ul>
  </div>
</DocumentFragment>
`;
